Tab stepper without navigation components support

SkyfaceD opened the pull request:
@acefalobi  What do you think about the feature where manage of views/fragments falls on the shoulders of the developer? I think it would be cool if the library allowed not to be tied to "Navigation Architecture Components", but to use it as a beautiful stepper view.

SkyfaceD committed:
Add example of tab stepper without navigation components support

acefalobi commented on app/src/main/java/com/aceinteract/android/stepper/presentation/samples/TabStepperWithoutNavigationComponentsActivity.kt:
```diff
@@ -0,0 +1,97 @@
+import androidx.core.view.isVisible
+import com.aceinteract.android.stepper.R
+import com.aceinteract.android.stepper.StepperNavListener
+import kotlinx.android.synthetic.main.tab_stepper_without_navigation_components_activity.*
```
Do you want to consider using viewbinding instead of kotlin synthetic here? Just for uniformity with the other activities.

SkyfaceD replied:
I can rewrite on viewBinding, but all other activities in samples package use kotlin synthetic.

acefalobi replied:
Oh haha. My bad. I was thinking of something else. Yeah, this is fine

acefalobi commented on app/src/main/java/com/aceinteract/android/stepper/presentation/samples/TabStepperWithoutNavigationComponentsActivity.kt:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    private fun setupPreviousButtonVisibility(step: Int) {
+        when (step > 0) {
```
You should just use a regular if/else statement here.

E.g:

```kotlin
if (step > 0) button_previous.show()
else button_previous.hide()
```

Or better still, just simplify to a single expression:

```kotlin
button_previous.isVisible = step > 0
```

SkyfaceD replied:
In my opinion when expression looks just cooler and on decompiled to java bytecode it's just regular if/else, I will replace it. isVisible doesn't provide zoomIn/zoomOut animation for fab, but I will replace it too like on other samples.

acefalobi replied:
It might be cooler, but `when` (in this case) uses more code for the same task.

You don't have to use `isVisible`, I just brought that up as an alternative for a simpler expression.

acefalobi commented on app/src/main/java/com/aceinteract/android/stepper/presentation/samples/TabStepperWithoutNavigationComponentsActivity.kt:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    private fun setupNextButtonImage(step: Int) {
+        when (step == 3) {
```
Here too

SkyfaceD replied:
Got it.

acefalobi commented on stepper/src/main/java/com/aceinteract/android/stepper/StepperNavigationView.kt:
```diff
@@ -224,6 +224,22 @@ class StepperNavigationView(context: Context, attrs: AttributeSet) : FrameLayout
+    /**
+     * Sets up the stepper to navigate steps without using navigation components.
+     */
+    fun setupWithoutNavController() {
```
There is no need for this, just move this to the original initialization for `onStepChanged` on line 126 and replace the when with if/else

Like this:
```kotlin
onStepChanged = {
    if (it == menu.size()) {
        stepperNavListener?.onCompleted()
    } else {
        menu.currentStep = it
        menu.updateUI()
        stepperNavListener?.onStepChanged(currentStep)
    }
}
```

SkyfaceD replied:
You are absolutely right.

acefalobi commented:
> @acefalobi What do you think about the feature where manage of views/fragments falls on the shoulders of the developer? I think it would be cool if the library allowed not to be tied to "Navigation Architecture Components", but to use it as a beautiful stepper view.

this is a really good idea. good work. just address the comments I made and rebase with the new master branch (I added a LICENSE file)

SkyfaceD commented:
> > @acefalobi What do you think about the feature where manage of views/fragments falls on the shoulders of the developer? I think it would be cool if the library allowed not to be tied to "Navigation Architecture Components", but to use it as a beautiful stepper view.
> 
> this is a really good idea. good work. just address the comments I made and rebase with the new master branch (I added a LICENSE file)

Okay.

SkyfaceD committed:
Update onStepChanged functionality in StepperNavigationView

SkyfaceD committed:
Fix TabStepperWithoutNavigationComponentsActivity logic

Replaced when expressions with if/else
Replaced fab show()/hide() with isVisible
Remove double click check, it makes no sense without animation

SkyfaceD committed:
Add TabStepperWithoutNavigationComponentsActivity to MainActivity list

SkyfaceD committed:
Merge remote-tracking branch 'upstream/master'
